temporalio/worker/worker.py

```diff
@@ -43,9 +45,12 @@ def __init__(
         *,
         task_queue: str,
         activities: Iterable[Callable] = [],
+        workflows: Iterable[Type] = [],
         activity_executor: Optional[concurrent.futures.Executor] = None,
+        workflow_task_executor: Optional[concurrent.futures.ThreadPoolExecutor] = None,
+        workflow_runner: WorkflowRunner = UnsandboxedWorkflowRunner(),
         interceptors: Iterable[Interceptor] = [],
-        max_cached_workflows: int = 0,
+        max_cached_workflows: int = 1000,
```

Comment:
This could probably be bumped much higher at least with the current implementation.

Reply:
I'd run some load tests and try to find a reasonable default.

Reply:
:+1: Will leave this as is until we start stressing the SDK then

Reply:
SGTM, might be linked to the workflow runner implementation too.
But I'm pretty sure this could be at least 10k by default. Leaving this up to you.

Reply:
I have opened #23